core/crypto/ecdh/ecdh.go

```diff
@@ -199,6 +199,22 @@ func (k *PrivateKey) UnmarshalBinary(data []byte) error {
 	return k.FromBytes(data)
 }
 
+// MarshalText is an implementation of a method on the
+// TextMarshaler interface defined in https://golang.org/pkg/encoding/
+func (k *PrivateKey) MarshalText() ([]byte, error) {
+	return []byte(base64.StdEncoding.EncodeToString(k.Bytes())), nil
+}
+
+// UnmarshalText is an implementation of a method on the
+// TextUnmarshaler interface defined in https://golang.org/pkg/encoding/
+func (k *PrivateKey) UnmarshalText(data []byte) error {
```

Comment:
maybe it'd be useful to put in the go doc that these deal with base64 ?

Reply:
Well okay then do we want to change it for all our crypto modules or just this one? And... can we do so in a separate ticket/PR?